src/main/java/edu/kit/provideq/toolbox/api/ProblemRouteDocumentation.java

```diff
@@ -0,0 +1,162 @@
+      ProblemManager<InputT, ResultT> manager) {
+    var exampleProblem = manager.getExampleInstances().stream()
+        .findFirst()
+        .orElseThrow(() -> new RuntimeException(
+            "No example available for problem " + manager.getType() + "!"));
+
+
+    var request = ProblemDto.fromProblem(exampleProblem);
+
+    String requestString;
+    try {
+      requestString = new ObjectMapper().writeValueAsString(request);
+    } catch (JsonProcessingException exception) {
+      throw new RuntimeException("no example available", exception);
+    }
+
+    var problemType = manager.getType();
+    return contentBuilder()
+        .example(exampleOjectBuilder()
+            .name(problemType.getId())
+            .value(requestString))
+        .schema(schemaBuilder()
+            .implementation(ProblemDto.class))
+        .mediaType(APPLICATION_JSON_VALUE);
+  }
+
+  private static org.springdoc.core.fn.builders.apiresponse.Builder buildProblemListResponse(
+      ProblemManager<?, ?> manager) {
+    var problemDtos = manager.getExampleInstances().stream()
+        .map(ProblemDto::fromProblem)
+        .toList();
+
+    String problemDtosJson;
+    try {
+      problemDtosJson = new ObjectMapper().writeValueAsString(problemDtos);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException("Could not serialize example problems!", e);
+    }
+
+    return responseBuilder()
+        .responseCode(String.valueOf(HttpStatus.OK.value()))
+        .content(contentBuilder()
+            .array(arraySchemaBuilder()
+                .schema(schemaBuilder().implementation(ProblemDto.class))
+            )
+            .example(exampleOjectBuilder()
+                .value(problemDtosJson)
+            )
+        );
+  }
+
+
+  private static org.springdoc.core.fn.builders.apiresponse.Builder buildProblemResponse(
+      ProblemManager<?, ?> problemManager) {
+    var contentBuilder = contentBuilder()
+        .mediaType(APPLICATION_JSON_VALUE)
+        .schema(schemaBuilder().implementation(ProblemDto.class));
+
+    // getExampleSolved, getExampleInvalid
+    int exampleProblemIndex = 1;
+    for (var exampleProblem : problemManager.getExampleInstances()) {
+      contentBuilder = contentBuilder.example(
+          buildExample(exampleProblem, "Example " + exampleProblemIndex)
+      );
+    }
+
+    return responseBuilder()
+        .responseCode(String.valueOf(HttpStatus.OK.value()))
+        .content(contentBuilder);
+  }
+
+  private static org.springdoc.core.fn.builders.exampleobject.Builder buildExample(
+      Problem<?, ?> exampleProblem, String name) {
+    String problemDtoJson;
+    try {
+      var problemDto = ProblemDto.fromProblem(exampleProblem);
+      problemDtoJson = new ObjectMapper().writeValueAsString(problemDto);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException("Example problem could not be parsed!", e);
+    }
+
+    return exampleOjectBuilder()
+        .name(name)
```

Comment:
No idea why swagger shows the name as the description... However, I don't think this generic description adds a lot of value since the problem type is redundant in this context. Maybe we can add names and descriptions to example problems at some point

Reply:
Yeah not a big deal.